pkg/apis/eventing/v1alpha1/channel_types.go

```diff
@@ -117,6 +117,26 @@ func (cs *ChannelStatus) GetCondition(t duckv1alpha1.ConditionType) *duckv1alpha
 	return chanCondSet.Manage(cs).GetCondition(t)
 }
 
+// IsReady returns true if the resource is ready overall.
```

Comment:
I understand you have just added some of this because of #515, but perhaps it is better to have them added in their own PR to  let this PR focus on Kafka.